src/dbus_api.rs

```diff
@@ -701,47 +687,35 @@ fn add_devs(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
     let dbus_context = m.path.get_data();
     let object_path = m.path.get_name();
     let return_message = message.method_return();
-    let return_sig = "(oqs)";
+    let return_sig = "s";
```

Comment:
This is a tricky one. We pass a list of paths as argument, although, as I've commented elsewhere, I think it should be a list of strings, so we can, for example, use the vpd83 information if it is available, instead of a path. And then, the engine figures out what those paths actually correspond to, and add some set of devices to the pool. There is no longer a 1-to-1 mapping between the values passed in and the actual devices used, as there may be duplicates and all sorts of other issues.

The best alternative return I have been able to come up with is a pair of values.
* a dict from device nodes actually added to sets of paths passed
* a vec of paths that didn't contribute to any of the device nodes because, e.g., a path actually corresponded to a device that was already in the pool

This is possible, because initialize() could return the mapping that could then be passed up to the ```add_*``` methods. In that case, the return value would have the signature ```(a{sas}as)```.

From the point of view of our stratis client, this could be nice. In some sort of verbose mode, it would actually display a little table of devices added and why, as well as of devices ignored.